include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1012 @@
+
+        @return Maximum number of decoded characters
+    */
+    BOOST_CONSTEXPR
+    size_type
+    max_size() const noexcept
+    {
+        return size_type( -1 );
+    }
+
+    /** Checks whether the string is empty
+
+        Checks if the string has no characters,
+        i.e. whether `size() == 0`.
+
+        @return `true` if the string is empty
+    */
+    BOOST_CONSTEXPR
+    bool
+    empty() const noexcept
+    {
+        return n_ == 0;
+    }
+
+    /** Copies characters
+
+        Copies a substring to the character array pointed to by
+        the destination char string, where `rcount` is the smaller
+        of `count` and `size() - pos`.
+
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid positions.
+
+        @throw out_of_range If `pos > size()`
+
+        @param dest pointer to the destination character string
+        @param count requested substring length
+        @param pos position of the first character
+
+        @return Number of characters copied
+    */
+    BOOST_URL_DECL
+    size_type
+    copy(char* dest,
+         size_type count,
+         size_type pos = 0) const;
+
+    /** Compares two strings
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(string_view other) const;
+
+    /// @copydoc compare(string_view)
+    int
+    compare(char const* other) const
+    {
+        return compare(string_view(other));
+    }
+
+    /** Compares two strings
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.to_string().data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(decoded_view other) const;
+
+    /** Returns a basic_string
+
+        This overload allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A>
+    std::basic_string<char, std::char_traits<char>, A>&
+    to_string(std::basic_string<char, std::char_traits<char>, A> & str) const
```

Comment:
we said we are going to do away with 1-arg `to_string` and just have `std::string to_string() const;`

Reply:
`assign_to` and `append_to` can both be used on recycled things